common/src/main/java/org/conscrypt/OpenSSLSocketImpl.java

```diff
@@ -16,1299 +16,118 @@
-     *         started.
-     */
-    public void setChannelIdPrivateKey(PrivateKey privateKey) {
-        if (!getUseClientMode()) {
-            throw new IllegalStateException("Server mode");
-        }
-
-        synchronized (stateLock) {
-            if (state != STATE_NEW) {
-                throw new IllegalStateException(
-                        "Could not change Channel ID private key after the initial handshake has"
-                                + " begun.");
-            }
-        }
-
-        if (privateKey == null) {
-            sslParameters.channelIdEnabled = false;
-            channelIdPrivateKey = null;
-        } else {
-            sslParameters.channelIdEnabled = true;
-            try {
-                ECParameterSpec ecParams = null;
-                if (privateKey instanceof ECKey) {
-                    ecParams = ((ECKey) privateKey).getParams();
-                }
-                if (ecParams == null) {
-                    // Assume this is a P-256 key, as specified in the contract of this method.
-                    ecParams =
-                            OpenSSLECGroupContext.getCurveByName("prime256v1").getECParameterSpec();
-                }
-                channelIdPrivateKey =
-                        OpenSSLKey.fromECPrivateKeyForTLSStackOnly(privateKey, ecParams);
-            } catch (InvalidKeyException e) {
-                // Will have error in startHandshake
-            }
-        }
-    }
-
-    @Override
-    public boolean getUseClientMode() {
-        return sslParameters.getUseClientMode();
-    }
-
-    @Override
-    public void setUseClientMode(boolean mode) {
-        synchronized (stateLock) {
-            if (state != STATE_NEW) {
-                throw new IllegalArgumentException(
-                        "Could not change the mode after the initial handshake has begun.");
-            }
-        }
-        sslParameters.setUseClientMode(mode);
-    }
-
-    @Override
-    public boolean getWantClientAuth() {
-        return sslParameters.getWantClientAuth();
-    }
-
-    @Override
-    public boolean getNeedClientAuth() {
-        return sslParameters.getNeedClientAuth();
+        super.setHostname(hostname);
     }
 
     @Override
-    public void setNeedClientAuth(boolean need) {
-        sslParameters.setNeedClientAuth(need);
-    }
-
-    @Override
-    public void setWantClientAuth(boolean want) {
-        sslParameters.setWantClientAuth(want);
-    }
-
-    @Override
-    public void sendUrgentData(int data) throws IOException {
-        throw new SocketException("Method sendUrgentData() is not supported.");
+    public String getHostnameOrIP() {
+        return super.getHostnameOrIP();
     }
 
     @Override
-    public void setOOBInline(boolean on) throws SocketException {
-        throw new SocketException("Methods sendUrgentData, setOOBInline are not supported.");
+    public SSLSession getHandshakeSession() {
+        return null;
```

Comment:
It will actually never return `null`. I've changed it to abstract here ... it's implemented by both subclasses.